samples/hello-cube.html

```diff
@@ -8,35 +8,39 @@
 <body>
 <canvas height=1000 width=1000></canvas>
 <script>
-/*******
-* This preview of Web GPU uses WHLSL as its shading language.
-* The choice of shading language(s) that will be ingested by the API is still under deliberation.
-*******/
 const positionAttributeNum  = 0;
 const colorAttributeNum     = 1;
 const transformBindingNum   = 0;
 const bindGroupIndex        = 0;
+const colorLocation         = 0;
 
 const shader = `
 struct FragmentData {
-    float4 position : SV_Position;
-    float4 color : attribute(${colorAttributeNum});
-}
+    [[builtin(position)]] position: vec4<f32>;
+    [[location(${colorLocation})]] color: vec4<f32>;
+};
+
+[[block]]
+struct Uniforms {
+    modelViewProjectionMatrix: mat4<f32>;
+};
+
+[[group(${bindGroupIndex}), binding(${transformBindingNum})]] var<uniform> uniforms;
 
-vertex FragmentData vertex_main(
-    float4 position : attribute(${positionAttributeNum}),
-    float4 color : attribute(${colorAttributeNum}),
-    constant float4x4[] modelViewProjectionMatrix : register(b${transformBindingNum}))
-{
+[[stage(vertex)]]
+fn main(
+    [[location(${positionAttributeNum})]] position: vec4<f32>
+    [[location(${colorAttributeNum})]] color: vec4<f32>,
+) -> FragmentData {
     FragmentData out;
-    out.position = mul(modelViewProjectionMatrix[0], position);
+    out.position = mul(uniforms.modelViewProjectionMatrix[0], position);
     out.color = color;
 
     return out;
 }
 
-fragment float4 fragment_main(float4 color : attribute(${colorAttributeNum})) : SV_Target 0
-{
+[[stage(fragment)]]
+fn main([[location(${colorLocation})]] color: vec4<f32>) -> [[location(0)]] vec4<f32> {
```

Comment:
This can't be called `main` as the vertex shader is called `main`.

Reply:
I thought we don't have this restriction. The entry point has to be uniquely identified by both the stage and the name, and it never participates in the name resolution within the module (since it can't be called), so there is no conflict there either.

Reply:
If memory serves, we removed that and just made them regular function names. So they follow the function naming restrictions, the stage doesn't come into account anymore.